server/src/main/java/org/elasticsearch/common/time/EpochMillisDateFormatter.java

```diff
@@ -52,8 +60,19 @@ public TemporalAccessor parse(String input) {
     }
 
     @Override
-    public DateFormatter withZone(ZoneId zoneId) {
-        return this;
+    public DateFormatter withZone(ZoneId newZoneId) {
```

Comment:
I'm wondering if anything other than UTC makes sense here, if not, would it make sense to either throw an UnsupportedOperationException and/or document this?